### PR TITLE
Improve Routing docs

### PR DESCRIPTION
**Pull request by afilina**
|    Q          |   A
|-------------- | ------
| Documentation | yes
| Bugfix        | no
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | no

### Description

Cleaning up the Laminas MVC routing documentation:

- Clean up the intro paragraphs.
- Give a simple but full config example using a Literal type.
- Explain in which file to put the routing config, since this is not obvious from any of the other pages.
- Make a bullet-point of the route types with a one-line explanation for each, then link to the appropriate heading in the [laminas-router/routing](https://docs.laminas.dev/laminas-router/routing/) docs. The idea is to avoid duplicating all the content from laminas-router in laminas-mvc.
- Note the version in which certain route types were deprecated and removed, in case someone lands here while upgrading an older application.
- Remove reference to laminas-console, since that package is no longer maintained.
- Simplify the note regarding last-in-first-out.

### CODE REVIEW

File: docs/book/routing.md [hunk 1]
```diff
@@ -1,725 +1,65 @@
...
+  the URI path. Example: `/blog` can have child routes with `/rss` and `/subscribe`, which would match `/blog/rss` and `/blog/subscribe` respectively.
+- [Regex](https://docs.laminas.dev/laminas-router/routing/#laminasrouterhttpplaceholder), available since version 3.2.0 of the router, matches a URI path using a regular expression. Example: `/product/(?<id>[0-9]+)` would match `/product/001`, with the `id` parameter containing `001`.
+- [Scheme](https://docs.laminas.dev/laminas-router/routing/#laminasrouterhttpscheme) matches the URI scheme. Example: `https`.
+- [Segment](https://docs.laminas.dev/laminas-router/routing/#laminasrouterhttpsegment) matching one or more segments of a URI path. Example: `/:blog/:article` would match `/blog/why-use-mvc`, with the `article` parameter containing `why-use-mvc`.
+- `Query` is deprecated since version 2.3.0 of the router and removed in version 3.0.0. You can now add query parameters without a query route.
+- `Wildcard` is deprecated since version 2.3.0 of the router and removed in version 3.0.0. Use the `Segment` route type instead.
```
Comment (froschdesign) [inline docs/book/routing.md]: The information about the availability of a route type should be covered in the laminas-router component, a short list is enough here on this page.
Deprecated types should no longer be used, so I would not list those here either.

Here is a preview of the current list:

Bildschirmfoto 2022-02-05 um 11 22 36

I think a simple line break would increase the overview:
Bildschirmfoto 2022-02-05 um 11 25 42

```suggestion
- [Hostname](https://docs.laminas.dev/laminas-router/routing/#laminasrouterhttphostname) matches domains and subdomains.  
  Example: `docs.laminas.dev`.
- [Literal](https://docs.laminas.dev/laminas-router/routing/#laminasrouterhttpliteral) matches an exact URI path.  
  Example: `/contact-us`
- [Method](https://docs.laminas.dev/laminas-router/routing/#laminasrouterhttpmethod) matches HTTP verbs.  
  Example: `post,put` for a route that submits a form.
- [Part](https://docs.laminas.dev/laminas-router/routing/#laminasrouterhttppart) allows crafting a tree of possible routes based on segments of
  the URI path.  
  Example: `/blog` can have child routes with `/rss` and `/subscribe`, which would match `/blog/rss` and `/blog/subscribe` respectively.
- [Regex](https://docs.laminas.dev/laminas-router/routing/#laminasrouterhttpplaceholder), matches a URI path using a regular expression.  
  Example: `/product/(?<id>[0-9]+)` would match `/product/001`, with the `id` parameter containing `001`.
- [Scheme](https://docs.laminas.dev/laminas-router/routing/#laminasrouterhttpscheme) matches the URI scheme.  
  Example: `https`.
- [Segment](https://docs.laminas.dev/laminas-router/routing/#laminasrouterhttpsegment) matching one or more segments of a URI path.  
  Example: `/:blog/:article` would match `/blog/why-use-mvc`, with the `article` parameter containing `why-use-mvc`.
```

Comment (froschdesign) [reply docs/book/routing.md]: Another option can be a definition list:
Bildschirmfoto 2022-02-05 um 11 28 40

```suggestion
[Hostname](https://docs.laminas.dev/laminas-router/routing/#laminasrouterhttphostname)
:   matches domains and subdomains.  
    Example: `docs.laminas.dev`.
[Literal](https://docs.laminas.dev/laminas-router/routing/#laminasrouterhttpliteral)
:   matches an exact URI path.  
    Example: `/contact-us`
[Method](https://docs.laminas.dev/laminas-router/routing/#laminasrouterhttpmethod)
:   matches HTTP verbs.  
    Example: `post,put` for a route that submits a form.
[Part](https://docs.laminas.dev/laminas-router/routing/#laminasrouterhttppart)
:   allows crafting a tree of possible routes based on segments of the URI path.  
    Example: `/blog` can have child routes with `/rss` and `/subscribe`, which would match `/blog/rss` and `/blog/subscribe` respectively.
[Regex](https://docs.laminas.dev/laminas-router/routing/#laminasrouterhttpplaceholder)
:   matches a URI path using a regular expression.
    Example: `/product/(?<id>[0-9]+)` would match `/product/001`, with the `id` parameter containing `001`.
[Scheme](https://docs.laminas.dev/laminas-router/routing/#laminasrouterhttpscheme)
:   matches the URI scheme.  
    Example: `https`.
[Segment](https://docs.laminas.dev/laminas-router/routing/#laminasrouterhttpsegment)
:   matching one or more segments of a URI path.  
    Example: `/:blog/:article` would match `/blog/why-use-mvc`, with the `article` parameter containing `why-use-mvc`.
```

See also: https://python-markdown.github.io/extensions/definition_lists/

Comment (afilina) [reply docs/book/routing.md]: I went for option 2: definition list.

**Review by Ocramius — APPROVED** [review]
Nice cleanup, thanks @afilina!

**Pull request merged**